rawsocket.c

```
@@ -111,28 +119,33 @@ static int raw_socket(void)
 }
 
 static PyObject *
-rawsocket_fd(PyObject *self, PyObject *args)
+rawsocket_fd(PyObject *self, PyObject *args, PyObject *kwargs)
 {
 	int fd;
-
-	if (!PyArg_ParseTuple(args, ""))
+	int family = AF_PACKET;
+	int protocol = ETH_P_ALL;
+	char* keywords[] = {"family", "protocol", NULL};
+	if (!PyArg_ParseTupleAndKeywords(args, kwargs, "|ii", keywords, &family, &protocol))
+	// if (!PyArg_ParseTuple(args, "ii", &family, &protocol))
 		return NULL;
 
-	fd = raw_socket();
+	fd = raw_socket(family, protocol);
 	if (fd < 0)
 		return NULL;
 
 	return PyInt_FromLong(fd);
 }
 
 PyDoc_STRVAR(rawsocket_fd_doc,
-"rawsocket_fd()\n\
+"rawsocket_fd(family=17, protocol=3)\n\
```

I don't like the magic numbers. so
`rawsocket_fd(family=socket.AF_PACKET, protocol=None)`

and keep a list of sane defaults for a certain address family inside `rawsocket_fd()`, where you can use ETH_P_ALL because you've imported net/ethernet.h.

Is there a sane default for AF_BLUEOOTH? BTPROTO_HCI maybe? I don't know. If there is no default, maybe raise an exception. But this behavior should be documented.

Also this function should validate the family to give some feedback to the user.